source/io/Stream.cpp

```diff
@@ -114,7 +114,7 @@ namespace Aws
                 auto read = m_stream->gcount();
                 buffer.len += static_cast<size_t>(read);
 
-                if (read > 0)
+                if (read > 0 || (read == 0 && m_stream->eof()))
```

Comment:
This was somehow pulled in from merged commit: https://github.com/awslabs/aws-crt-cpp/commit/8e4da9be463cd74d1a9f9f7ff4f113f3b85c4c9a